diff --git a/dist/preset.js b/dist/preset.js
index f8c5dab..302c1c9 100644
--- a/dist/preset.js
+++ b/dist/preset.js
@@ -51,7 +51,7 @@ async function vueComponentMeta(engine, tsconfigPath = "tsconfig.json") {
             let s = new MagicString(src), localBindings = await parseLocalBindings(id, src), sfcMainImportSource = (await parseDefaultImports(id, src)).get("_sfc_main"), sfcMainQuery = new URLSearchParams(sfcMainImportSource?.split("?")[1]), hasImportedSfcMain = id.endsWith(".vue") && sfcMainQuery.has("vue") && sfcMainQuery.get("type") === "script";
             return metaSources.forEach((meta) => {
               let isDefaultExport = meta.exportName === "default", name = isDefaultExport ? "_sfc_main" : meta.exportName;
-              !localBindings.has(name) && !(isDefaultExport && hasImportedSfcMain) || (!id.endsWith(".vue") && isDefaultExport && (s.replace("export default ", "const _sfc_main = "), s.append(`
+              !localBindings.has(name) && !(isDefaultExport && hasImportedSfcMain) && !(!id.endsWith(".vue") && isDefaultExport) || (!id.endsWith(".vue") && isDefaultExport && (s.replace("export default ", "const _sfc_main = "), s.append(`
 export default _sfc_main;`)), s.append(`
 ;${name}.__docgenInfo = Object.assign({
             displayName: ${name}.name ?? ${name}.__name
